Sources/Utils.php

```diff
@@ -998,6 +1004,81 @@ public static function ucwords(string $string): string
 		return self::convertCase($string, 'ucwords');
 	}
 
+	/**
+	 * Formats a string for use as a URL slug for queryless URLs.
+	 *
+	 * @param string $string The original string.
+	 * @param int $max_length Maximum length of the slug string. Default: 30.
+	 * @return string The slug string.
+	 */
+	public static function makeSlug(string $string, int $max_length = 30): string
+	{
+		static $common_words_regex;
+
+		// Decode any percent-encoding.
+		$slug = rawurldecode($string);
+
+		// Decode any HTML entities.
+		$slug = self::entityDecode($slug, true);
+
+		// Convert to lower case and compatibility decomposition form.
+		$slug = self::convertCase($slug, 'lower', false, 'kd', true);
+
+		// Remove common words.
+		if (!isset($common_words_regex)) {
+			Lang::load('Search');
+
+			$common_words_regex =
+				'/\b' .
+				self::buildRegex(
+					array_map(
+						fn($word) => self::convertCase($word, 'lower', false, 'kd', true),
+						explode(',', Lang::getTxt('search_stopwords')),
+					),
+					'/',
+				) .
+				'\b/u';
+		}
+
+		$slug = preg_replace($common_words_regex, ' ', $slug);
+
+		// Build the slug.
+		$slug = preg_replace_callback_array(
+			[
+				// Remove apostrophes, quotation marks, etc., within words.
+				'/(?<=[\p{L}\p{M}\p{N}])["\'\p{Pi}\p{Pf}](?=[\p{L}\p{M}\p{N}])/u' => fn($matches) => '',
+
+				// Remove combining and enclosing marks.
+				'/[\p{Mn}\p{Me}]/u' => fn($matches) => '',
+
+				// Replace all characters that aren't letters or numbers with hyphens.
+				// Note: ideographic characters count as letters.
+				'/[^\p{L}\p{N}]/u' => fn($matches) => '-',
+
+				// Collapse runs of hyphens.
+				'/-+/u' => fn($matches) => '-',
+
+				// Trim leading and trailing hyphens.
+				'/^-|-$/u' => fn($matches) => '',
+			],
```

Comment:
This can be `preg_replace;() ` as it accepts arrays. The Unicode flag is not required for the last two patterns.

Reply:
Both true